Orange/widgets/visualize/pythagorastreeviewer.py

```diff
@@ -0,0 +1,759 @@
+        return self.boundingRect().size() + \
+               QtCore.QSizeF(self._padding, self._padding)
+
+
+class SquareGraphicsItem(QtGui.QGraphicsRectItem):
+    """Square Graphics Item.
+
+    Square component to draw as components for the non-interactive Pythagoras
+    tree.
+
+    Parameters
+    ----------
+    tree_node : TreeNode
+        The tree node the square represents.
+    brush : QColor, optional
+        The brush to be used as the backgound brush.
+    pen : QPen, optional
+        The pen to be used for the border.
+
+    """
+
+    def __init__(self, tree_node, parent=None, **kwargs):
+        self.tree_node = tree_node
+        self.tree_node.graphics_item = self
+
+        center, length, angle = tree_node.square
+        self._center_point = center
+        self.center = QtCore.QPointF(*center)
+        self.length = length
+        self.angle = angle
+        super().__init__(self._get_rect_attributes(), parent)
+        self.setTransformOriginPoint(self.boundingRect().center())
+        self.setRotation(degrees(angle))
+
+        self.setBrush(kwargs.get('brush', QtGui.QColor('#297A1F')))
+        self.setPen(kwargs.get('pen', QtGui.QPen(QtGui.QColor('#000'))))
+
+        self.setAcceptHoverEvents(True)
+        self.setZValue(kwargs.get('zvalue', 0))
+        self.z_step = Z_STEP
+
+        # calculate the correct z values based on the parent
+        if self.tree_node.parent != -1:
+            p = self.tree_node.parent
+            # override root z step
+            num_children = len(p.children)
+            own_index = [1 if c.label == self.tree_node.label else 0
+                         for c in p.children].index(1)
+
+            self.z_step = int(p.graphics_item.z_step / num_children)
+            base_z = p.graphics_item.zValue()
+
+            self.setZValue(base_z + own_index * self.z_step)
+
+    def _get_rect_attributes(self):
+        """Get the rectangle attributes requrired to draw item.
+
+        Compute the QRectF that a QGraphicsRect needs to be rendered with the
+        data passed down in the constructor.
+
+        """
+        height = width = self.length
+        x = self.center.x() - self.length / 2
+        y = self.center.y() - self.length / 2
+        return QtCore.QRectF(x, y, height, width)
+
+
+class InteractiveSquareGraphicsItem(SquareGraphicsItem):
+    """Interactive square graphics items.
+
+    This is different from the base square graphics item so that it is
+    selectable, and it can handle and react to hover events (highlight and
+    focus own branch).
+
+    Parameters
+    ----------
+    tree_node : TreeNode
+        The tree node the square represents.
+    brush : QColor, optional
+        The brush to be used as the backgound brush.
+    pen : QPen, optional
+        The pen to be used for the border.
+
+    """
+
+    timer = QtCore.QTimer()
+
+    MAX_OPACITY = 1.
+    SELECTION_OPACITY = .5
+    HOVER_OPACITY = .1
```

Comment:
I made it that way because for larger trees, where there is a lot of overlap (e.g. *adult* with sqrt sizing) the almost invisible-ness of other branches makes the branch you are hovering over very clear. If I change the hover opacity to anything larger (e.g. 0.2), it becomes less clear which branch is being highlighted.

If you feel this is unnecessary and having a higher opacity would be better I can change this.


[0.1 opacity](https://i.gyazo.com/5b5dcdd53df161b840e1e4b92e3ffcdc.png) vs [0.2 opacity](https://i.gyazo.com/2f1bf2a4015435664323eea0f556c6a3.png)